srcs/runners_manager/vm_creation/aws/AwsManager.py

```diff
@@ -0,0 +1,142 @@
+                SubnetId=runner.vm_type.config["subnet_id"],
+                MaxCount=1,
+                MinCount=1,
+                UserData=user_data,
+                BlockDeviceMappings=[
+                    {
+                        'DeviceName': '/dev/sda1',
+                        'Ebs': {
+                            'VolumeSize': int(runner.vm_type.config["disk_size_gb"]),
+                            'DeleteOnTermination': True,
+                            'VolumeType': 'gp2'
+                        },
+                    },
+                ],
+                TagSpecifications=[
+                    {
+                        'ResourceType': 'instance',
+                        'Tags': [
+                            {
+                                'Key': 'Name',
+                                'Value': runner.name
+                            },
+                            {
+                                'Key': 'Runner-manager',
+                                'Value': 'true'
+                            }
+                        ]
+                    },
+                ],
+            )
+            
+            return instance['Instances'][0]['InstanceId']
+
+
+        except Exception as exception:
+            metrics.runner_creation_failed.labels(cloud=self.name).inc()
+            logger.error(exception)
+            raise exception
+
+    @delete_vm_metric
+    def delete_vm(self, runner):
+        try:
+            response = self.ec2.describe_instances(InstanceIds=[runner.vm_id])
+            tags = response['Reservations'][0]['Instances'][0]['Tags']
+            for tag in tags:
+                if tag['Key'] == 'Name' and tag.get('Value') == {runner.name}:
+                    self.ec2.terminate_instances(InstanceIds=[runner.vm_id])
+                    logger.info(f"Instance with ID {runner.vm_id} terminated.")
+                    break
+
+            logger.info(f"Instance of runner {runner.name} has been terminated")
+
+        except Exception as exception:
+            logger.info(f"Instance of runner {runner.name} {exception}")
+            pass
+
+    def get_all_vms(self, prefix: str) -> list[Runner]:
+        # Get all vms in ec2
+        instances = self.ec2.describe_instances()
+        runners = []
+        for instance in instances:
+            for tag in instance.get('Tags', []):
+                if tag.get('Key') == 'Name' and tag.get('Value').startswith(prefix):
+                    runner = Runner(
+                        tag.get('Value'),
+                        instance.get('InstanceId'),
+                        VmType(
+                            {
+                                "tags": [],
+                                "config": {},
+                                "quantity": {},
+                            }
+                        ),
+                        self.name
+                    )
+                    runners.append(runner)
+            return runners
```

Comment:
Have you tried using `items()` when looping through tags? if it works I like this writing a bit more.

```suggestion
            for key, value in instance.get('Tags', []).items():
                if key == 'Name' and value.startswith(prefix):
                    runner = Runner(
                        value,
                        instance.get('InstanceId'),
                        VmType(
                            {
                                "tags": [],
                                "config": {},
                                "quantity": {},
                            }
                        ),
                        self.name
                    )
                    runners.append(runner)
            return runners
```

Reply:
Damn, I didn't mention the reason. You can't use the `.items()` method on `tags`! It triggers an error, so I leave it as before